onset_maproom/tests/test_calc.py

```diff
@@ -37,3 +496,90 @@ def test_onset():
     # assertion, but gives a more helpful error message when the test
     # fails: Timedelta('6 days 00:00:00')
     # vs. numpy.timedelta64(518400000000000,'ns')
+
+
+def test_onset_date_returns_nat():
+
+    precip = set_precip_sample()
+    precipNaN = precip + np.nan
+    onsetsNaN = calc.onset_date(
+        daily_rain=precipNaN,
+        wet_thresh=1,
+        wet_spell_length=3,
+        wet_spell_thresh=20,
+        min_wet_days=1,
+        dry_spell_length=7,
+        dry_spell_search=21,
+    )
+    assert np.isnat(onsetsNaN.values)
+
+
+def test_onset_date_dry_spell_invalidates():
+
+    precip = set_precip_sample()
+    precipDS = xr.where(
+        (precip["T"] > pd.to_datetime("2000-05-09"))
+        & (precip["T"] < (pd.to_datetime("2000-05-09") + pd.Timedelta(days=5))),
+        0,
+        precip,
+    )
+    onsetsDS = calc.onset_date(
+        daily_rain=precipDS,
+        wet_thresh=1,
+        wet_spell_length=3,
+        wet_spell_thresh=20,
+        min_wet_days=1,
+        dry_spell_length=7,
+        dry_spell_search=21,
+    )
+    assert pd.Timedelta(onsetsDS.values) != pd.Timedelta(days=6)
+
+
+def test_onset_date_late_dry_spell_invalidates_not():
+
+    precip = set_precip_sample()
+    preciplateDS = xr.where(
+        (precip["T"] > (pd.to_datetime("2000-05-09") + pd.Timedelta(days=20))),
+        0,
+        precip,
+    )
+    onsetslateDS = calc.onset_date(
+        daily_rain=preciplateDS,
+        wet_thresh=1,
+        wet_spell_length=3,
+        wet_spell_thresh=20,
+        min_wet_days=1,
+        dry_spell_length=7,
+        dry_spell_search=21,
+    )
+    assert pd.Timedelta(onsetslateDS.values) == pd.Timedelta(days=6)
+
+
+def test_onset_date_1st_wet_spell_day_not_wet_day():
+    """May 4th is 0.28 mm thus not a wet day
+    resetting May 5th and 6th respectively to 1.1 and 18.7 mm
+    thus, May 5-6 are both wet days and need May 4 to reach 20mm
+    but the 1st wet day of the spell is not 4th but 5th
+    """
+
+    precip = set_precip_sample()
+    precipnoWD = xr.where(
+        (precip["T"] == pd.to_datetime("2000-05-05")),
+        1.1,
+        precip,
+    )
+    precipnoWD = xr.where(
+        (precip["T"] == pd.to_datetime("2000-05-06")),
+        18.7,
+        precipnoWD,
+    )
```

Comment:
Indeed this answers my question. (My eye had failed to pick up that 05-05 and 05-06 were different dates.)

I think there's probably a way to change both elements in a single call to `where`. I don't know it off the top of my head.